keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/DeliveryConfigRepository.kt

```diff
@@ -20,25 +21,19 @@ interface DeliveryConfigRepository : PeriodicallyCheckedRepository<DeliveryConfi
    */
   fun get(name: String): DeliveryConfig
 
-  /**
-   * Check if the given application has a [DeliveryConfig].
-   * @return Boolean
-   */
-  fun hasDeliveryConfig(application: String): Boolean
-
   /**
    * Retrieve the [Environment] a resource belongs to, by the resource [id].
    *
    * @return An [Environment] or `null` if the resource is not managed via an [Environment]
    */
-  fun environmentFor(resourceId: ResourceId): Environment?
+  fun environmentFor(resourceId: ResourceId): Environment
 
   /**
    * Retrieve the [DeliveryConfig] a resource belongs to (the parent of its environment).
    *
    * @return A [DeliveryConfig] or `null` if the resource is not managed via a [DeliveryConfig].]
    */
-  fun deliveryConfigFor(resourceId: ResourceId): DeliveryConfig?
+  fun deliveryConfigFor(resourceId: ResourceId): DeliveryConfig
```

Comment:
Same as ^^^

Reply:
I'll update. Yes it does: an `OrphanedResourceException`